### PR TITLE
Fix/litellm cerebras groq compatibility

### PR DESCRIPTION
**Pull request by aditya270520**
## 🔧 Solution
   
   - **Override message formatting** in `LiteLLMModel` to ensure compatibility with LiteLLM providers
   - **Add `_format_request_message_contents` method** for LiteLLM-compatible content formatting
   - **Add `_format_request_messages` method** to override parent class behavior
   - **Update `format_request` and `structured_output` methods** to use new formatting
   - **Maintain backward compatibility** with existing functionality
   
   ## ✅ What This Fixes
   
   - **Cerebras models** (e.g., `cerebras/qwen-3-235b-a22b-instruct-2507`)
   - **Groq models** (e.g., `groq/llama3-70b-8192`)
   - **All other LiteLLM providers** continue to work as before
   
   ## 🧪 Testing
   
   - **Unit tests updated** to reflect new message format
   - **Comprehensive edge case testing** completed locally
   - **All tests pass** including performance and memory tests
   - **No regressions** introduced
   
   ## �� Technical Details
   
   The fix ensures that simple text content is formatted as strings rather than lists of content blocks, which is required by certain LiteLLM providers like Cerebras and Groq.
   
   **Before (Broken)**:
   ```python
   "messages": [{"role": "user", "content": [{"text": "What is 2+2", "type": "text"}]}]
   ```
   
   **After (Fixed)**:
   ```python
   "messages": [{"role": "user", "content": "What is 2+2"}]
   ```
   
   ## �� Files Changed
   
   - `src/strands/models/litellm.py` - Core fix implementation
   - `tests/strands/models/test_litellm.py` - Updated unit tests

New feature
Breaking change
Documentation update
Other (please describe):

## Testing

How have you tested the change?  Verify that the changes do not break functionality or introduce warnings in consuming repositories: agents-docs, agents-tools, agents-cli

- [ ] I ran `hatch run prepare`

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [x] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/strands/models/litellm.py [hunk 1]
```diff
@@ -103,6 +103,127 @@ def format_request_message_content(cls, content: ContentBlock) -> dict[str, Any]
 
         return super().format_request_message_content(content)
 
+    def _format_request_message_contents(self, role: str, content: ContentBlock) -> list[dict[str, Any]]:
+        """Format LiteLLM compatible message contents.
+
+        LiteLLM expects content to be a string for simple text messages, not a list of content blocks.
+        This method flattens the content structure to be compatible with LiteLLM providers like Cerebras and Groq.
```
Comment (dbschmigelski) [inline src/strands/models/litellm.py:115]: Hi thanks for this but can you help explain where the abstraction is broken.

Are you indicating that LiteLLM has a broken abstraction for Cerebras and Groq? Or do you believe that Strands has always improperly implemented the spec but this was just exposed when Ceregras and Groq were added?

Comment (dbschmigelski) [reply src/strands/models/litellm.py:115]: bump @aditya270520, thanks

Comment (aditya270520) [reply src/strands/models/litellm.py:115]: LiteLLM has a broken abstraction for Cerebras and Groq providers. Strands has always properly implemented the OpenAI specification, but this inconsistency was exposed when Cerebras and Groq were added to LiteLLM's supported providers.
The abstraction should be fixed in LiteLLM itself to ensure all providers receive content in the same format, but until that happens, Strands needs this workaround to maintain compatibility.
This is a common issue when trying to create unified interfaces across multiple providers with different underlying APIs - the abstraction layer (LiteLLM) needs to handle the normalization, but it's not doing so consistently.

Comment (aditya270520) [reply src/strands/models/litellm.py:115]: https://github.com/strands-agents/sdk-python/issues/729#issuecomment-3314723770
